centreon/www/api/class/centreon_realtime_services.class.php

```
@@ -564,6 +564,35 @@ protected function setServiceFieldList()
+            'last_hard_state_change' => ['s.last_hard_state_change', 'h.name', 's.description'],
+            'last_check' => ['s.last_check', 'h.name', 's.description'],
+            'current_attempt' => ['s.check_attempt', 'h.name', 's.description'],
+            'output' => ['s.output', 'h.name', 's.description'],
+            'default' => ['s.description', 'h.name'],
+        ];
+        $sortType = $this->sortType ?? 'default';
+        $columnsToAdd = $sortColumns[$sortType] ?? $sortColumns['default'];
+        foreach ($columnsToAdd as $col) {
+            // If column isn't already in the SELECT, add it
+            $alreadyPresent = false;
+            foreach ($fields as $key => $alias) {
```

PHPMD (2.15.0)

Avoid unused local variables such as '$alias'. (undefined)

(UnusedLocalVariable)